rest_authenticate/services/session.py

```diff
@@ -0,0 +1,55 @@
+import datetime
+
+from odoo import fields
+from odoo.http import request, root
+from odoo.service import security
+
+from odoo.addons.base_rest import restapi
+from odoo.addons.component.core import Component
+
+
+def _rotate_session(httprequest):
+    if httprequest.session.rotate:
+        root.session_store.delete(httprequest.session)
+        httprequest.session.sid = root.session_store.generate_key()
+        if httprequest.session.uid:
+            httprequest.session.session_token = security.compute_session_token(
+                httprequest.session, request.env
+            )
+        httprequest.session.modified = True
+
+
+class SessionAuthenticateService(Component):
+    _inherit = "base.rest.service"
+    _name = "session.authenticate.service"
+    _usage = "authenticate"
+    _collection = "session.rest.services"
+
+    @restapi.method([(["/"], "POST")])
```

Comment:
shouldn't be `/authenticate`?

Reply:
No, because `_usage` is already at the end of the url.

Reply:
yeah, I know, but it seems weird to have only the root path there.

@lmignon what would be the recommended way in this case?
I'd say that if you specify the route, you should take full control of it. It would also make clear which endpoint you are defining.